L1Trigger/Phase2L1ParticleFlow/plugins/L1JUMPProducer.cc

```diff
@@ -0,0 +1,153 @@
+#include <vector>
+#include <string>
+#include <ap_int.h>
+#include <ap_fixed.h>
+#include <TVector2.h>
+#include <iostream>
+
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/L1TParticleFlow/interface/PFCandidate.h"
+#include "DataFormats/L1TParticleFlow/interface/PFJet.h"
+
+#include "DataFormats/L1TParticleFlow/interface/puppi.h"
+#include "DataFormats/L1TParticleFlow/interface/sums.h"
+#include "DataFormats/L1TParticleFlow/interface/jets.h"
+#include "DataFormats/L1TParticleFlow/interface/layer1_emulator.h"
+
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+
+#include "L1Trigger/Phase2L1ParticleFlow/interface/jetmet/L1PFJUMPEmulator.h"
+
+using namespace l1t;
+
+class L1JUMPProducer : public edm::global::EDProducer<> {
+  // JUMP Producer
+  // JUMP: Jet Uncertainty-aware MET Prediction
+public:
+  explicit L1JUMPProducer(const edm::ParameterSet&);
+  ~L1JUMPProducer() override;
+
+private:
+  void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const override;
+  edm::EDGetTokenT<std::vector<l1t::EtSum>> metToken;
+  edm::EDGetTokenT<std::vector<l1t::PFJet>> jetsToken;
+
+  typedef ap_ufixed<14, 12, AP_TRN, AP_SAT> pt_t;  // LSB = 0.25 GeV
+  typedef ap_ufixed<28, 24, AP_TRN, AP_SAT> pt2_t;
```

Comment:
These could be used from `DataFormats/L1TParticleFlow/interface/datatypes.h` as `l1ct::pt_t` and `l1ct::pt2_t`